### PR TITLE
Sils/iglobby

### PR DESCRIPTION
**Pull request by sils**
### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [ ] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [ ] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [ ] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

**After you submit your pull request, DO NOT click the 'Update Branch' button**

### CODE REVIEW

**Comment by sils** [comment]
CC @underyx this is your change but using lru cache and affecting all usages, I think it's a bit cleaner. Special thanks to @userzimmermann who discussed until we found that solution.

CC @jayvdb to make sure I'm deprecating things properly ;)

File: coalib/parsing/Globbing.py [hunk 1]
```diff
@@ -388,13 +388,28 @@ def iglob(pattern):
...
         else:
-            for name in _iglob(pat):
+            for name in _simple_glob(pat):
                 yield name.rstrip(os.sep)
 
 
+def iglob(pattern):
+    """
+    DEPRECATED. Do not use. Wraps glob.
+
```
Comment (userzimmermann) [inline coalib/parsing/Globbing.py]: @sils Why not really mark as deprecated? By raising a deprecation warning. Or is there a decorator?

Comment (sils) [inline coalib/parsing/Globbing.py]: we don't have a decorator for that yet

**Comment by underyx** [comment]
@sils I actually tried this last week myself as well. Somehow I didn't see the same performance increase as with the other method. Have you benchmarked this?

**Comment by sils** [comment]
Not yet @userzimmermann wanted to. Maybe he has?

**Comment by sils** [comment]
closing as @underyx ' solution seems to be better

**Pull request closed without merging**